scripts/gen_openapi.sh

```diff
@@ -9,6 +9,9 @@ set -e
 #   1. Vault has been checked out at an appropriate version and built
 #   2. vault executable is in your path
 #   3. Vault isn't already running
+#   4. Jq is installed
+
+cd "$(dirname "${BASH_SOURCE[0]}")"
```

Comment:
`cd` is usually not a good idea for shell scripts. If the directory doesn't exist, the subsequent operations will be done on the wrong directory.

Reply:
Hmm. This was the recommendation to make the relative path immune to the run location. It seems like BASH_SOURCE should always exist b/c it's where you're running the script from. But I could be misunderstanding this

Reply:
I'm not really familiar with `BASH_SOURCE` so I might be wrong here. Would it be possible to instead prefix every path with `"${BASH_SOURCE[0]}"`

Reply:
hmmm. lemme try it

Reply:
so, `BASH_SOURCE` is the path to the script file. It will always exist. So, I could change it for convention purposes, but this seems to be pretty common practice. I could change it for convention, I'm just not super familiar with shell script convention.

Reply:
OK, if this always works then it should be good 👍